### ISSUE TITLE
Throw when image field is null

### ISSUE THREAD

**Issue by Kerumen** [graysonhicks/gatsby-plugin-remote-images#39]
I'm using this plugin with a remote CMS and sometimes my image field is not set. 
Unfortunately my build throws with:

```
Cannot read property 'localImage' of undefined
```

I searched similar plugins to see different implementations and I've seen this one: [gatsby-source-sanity-transform-images](https://github.com/reactgraphqlacademy/gatsby-source-sanity-transform-images) where they use `setFieldsOnGraphQLNodeType` to add the field.

Maybe this plugin should do the same to avoid that kind of errors?

**Comment by Kerumen** [comment]
FWIW, I've seen [this issue](https://github.com/graysonhicks/gatsby-plugin-remote-images/issues/37#issuecomment-581484570) and tried to revert the version to `1.0.7` and it works.

**Comment by fatihky** [comment]
Hi, I've also encountered with the same error. This line causes the error:
https://github.com/graysonhicks/gatsby-plugin-remote-images/blob/6ffa5293f80d322a65b0470e9bfc3fa4948b1cc2/src/gatsby-node.js#L144

**Comment by mjsarfatti** [comment]
Same problem here. I have a custom source plugin that pulls data from an API, and sometimes the `image` field is `null`.

Reverting to version `1.0.7` works.

**Comment by chandlervdw** [comment]
@Kerumen are you using this with sanity.io? If so, could you shed some light on your `gatsby-config` concerning this plugin's options?

**Comment by ooloth** [comment]
I'm seeing the same issue when the remote response is a temporary `502`. (The build breaks entirely with multiple errors messages like the one in the description above.)

Reverting to `1.0.7` works for me as well.

**Comment by graysonhicks** [comment]
Gave this a go today and have an update. According [this page in the docs](https://www.gatsbyjs.org/docs/schema-gql-type/), the `setFieldsOnGraphQLNodeType` has been deprecated due to the new `createTypes` action.

I attempted using the `createTypes` API, but I still get the `"Cannot read property 'localImage' of undefined"` error when querying the node type when no nodes have been created.

Here is what I added to `gatsby-node`:
```javascript
exports.createSchemaCustomization = ({ actions, schema }, options) => {
  const { createTypes } = actions;
  const { nodeType, name = 'localImage' } = options;
  const typeDefs = [
    schema.buildObjectType({
      name: nodeType,
      fields: {
        // Also need this to handle an array of Files, but not sure how to get a union here.
        [name]: 'File',
      },
      interfaces: ['Node'],
    }),
  ];
  createTypes(typeDefs);
};
```

**WIP branch**: `feature/throw-when-image-null`

cc: @Kerumen

**Comment by amitash** [comment]
What is the solution to this issue?

**Comment by andyaaz** [comment]
I was also experiencing the same thing. I was able to fix it by creating a fake file node when url is invalid. quite hacky but I don't know how else to fix it. my branch: https://github.com/andyaaz/gatsby-plugin-remote-images/tree/fix/image-is-nullish

**Comment by sthomp** [comment]
thanks @andyaaz . Using your branch solved the issue for me.

**Comment by GreatState** [comment]
any chance that @andyaaz change will be merged into the main plugin?

**Comment by graysonhicks** [comment]
Opened [PR here](#67).  Need to review and fix conflicts.  I do feel like there is likely a less hacky way to handle this, but happy it seems to work for people! 😅

**Comment by andyaaz** [comment]
@graysonhicks resolved conflicts and added tests

**Comment by graysonhicks** [comment]
Wow, this took me forever and I forgot about it.  Merged and released with `3.4.0`.  Thanks!
